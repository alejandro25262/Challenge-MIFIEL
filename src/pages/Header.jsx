import React from "react";
import { useSelector } from "react-redux";
import Logo from "./components/Logo";
import Avatar from "./components/Avatar";

const Header = () => {
  const { user } = useSelector((state) => state);
  const { loading, name, email } = user;

  return (
    <div className="w-full h-20 px-40 flex justify-between items-center border-gray-900 border-b-2">
      <Logo />
      {!loading ? (
        <div className="flex justify-center items-center">
          {`${name}(${email})`}
          <Avatar className="bg-gray-900 text-white ml-2" />
        </div>
      ) : (
        <p>Loading</p>
      )}
    </div>
  );
};

export default Header;
