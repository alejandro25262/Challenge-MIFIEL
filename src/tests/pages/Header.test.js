import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import * as reactRedux from "react-redux";
import Header from "../../pages/Header";

describe("<Header />", () => {
  const mockStore = configureStore();
  const initialState = {
    user: { name: "Jesus", email: "a@a.com", avatar: "JL" },
  };

  test("match snapshot", () => {
    const store = mockStore(initialState);
    jest
      .spyOn(reactRedux, "useSelector")
      .mockImplementation((state) => store.getState());

    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
