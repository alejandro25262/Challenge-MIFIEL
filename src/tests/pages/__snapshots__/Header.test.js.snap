// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> match snapshot 1`] = `
<div
  className="w-full h-20 px-40 flex justify-between items-center border-gray-900 border-b-2"
>
  <Logo />
  <div
    className="flex justify-center items-center"
  >
    Jesus(a@a.com)
    <Avatar
      className="bg-gray-900 text-white ml-2"
    />
  </div>
</div>
`;
